Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Auth/Register/RegisterViewController.swift

```diff
@@ -64,20 +64,30 @@ class RegisterViewController: UIViewController {
             if isValidEmail(emailID: userEmail ?? "") == false {
                 alertMessage(userMessage: "Please enter valid email address!")
                 view2.titleLabel.textColor = .red
+                view3.titleLabel.textColor = UIColor.colorText
                 view5.titleLabel.textColor = UIColor.colorText
             } else {
-                // check for password validation
-                // minimum eight characters, at least one letter, one number and one special character
-                if isValidPassword(password: userPassword ?? "") == false {
-                    alertMessage(userMessage: "Please enter a valid password! (min 8 characters, one letter, one number and one special character)")
-                    view2.titleLabel.textColor = UIColor.colorText
-                    view5.titleLabel.textColor = .red
-                } else {
-                    // register data is all good
+                if isValidID(personalID: userPersonalID ?? "") == false {
+                    alertMessage(userMessage: "Please enter a correct and unique ID")
                     view2.titleLabel.textColor = UIColor.colorText
+                    view3.titleLabel.textColor = .red
                     view5.titleLabel.textColor = UIColor.colorText
-                    let user = User(email: userEmail, name: userName, personalID: userPersonalID, studentID: userStudentID, password: userPassword, photo: "")
-                    registerUser(user: user)
+                } else {
+                    // check for password validation
+                    // minimum eight characters, at least one letter, one number and one special character
+                    if isValidPassword(password: userPassword ?? "") == false {
+                        alertMessage(userMessage: "Please enter a valid password! (min 8 characters, one letter, one number and one special character)")
+                        view2.titleLabel.textColor = UIColor.colorText
+                        view3.titleLabel.textColor = UIColor.colorText
+                        view5.titleLabel.textColor = .red
+                    } else {
+                        // register data is all good
+                        view2.titleLabel.textColor = UIColor.colorText
+                        view3.titleLabel.textColor = UIColor.colorText
+                        view5.titleLabel.textColor = UIColor.colorText
+                        let user = User(email: userEmail, name: userName, personalID: userPersonalID, studentID: userStudentID, password: userPassword, photo: "")
+                        registerUser(user: user)
+                        }
```

Comment:
It looks like a lot of duplicate code. Can you find a way to avoid/mitigate this?